### DISCUSSION TITLE
How to redirect an audio from a file into my mic?

### DISCUSSION THREAD

**Discussion by smktpd** [ExistentialAudio/BlackHole#809] [Q&A] [answered]
I have a laugh.wav file, built-in mic and built-in speakers.

I run a browser (Thorium) to open a website for tele-video-conferences.

I would like to be able to play laugh.wav for others to hear in the conference, but I don't want it to play it on my speakers (because otherwise I wouldn't even need blackhole, I'd just play file to built-in speakers while unmuted so that built-in mic would capture it from the speakers (mixed with other background noise it'd capture from my environment).

I did this:
1. created a Multi-Output Device with 'built-in speakers' going first (and used as primary device) and 'Blackhole 2ch' going 2nd. I've set that device as default output.
2. created an Aggregate Device with 'built-in mic' going first (and used as primary device) and 'Blackhole 2ch' going 2nd. I've set that device as default input.
3. I've hit checkbox 'mute' for 'built-in mic'
4. Opened the tele-video-conference site in the browser, unmuted there.
5. I tried to use `mpv` with each existing value of `--audio-device` - and some of them either play it to my 'built-in speakers' (so I can hear it), some play it without me hearing the output, some seem to just error out (the player doesn't show the playback progress timer).
As a result - the others can't hear the audio from my file.

What am I doing wrong?

**Comment by devinroth** [comment, marked as answer]
First off, you are misunderstanding what an aggregate device does. It does not mix the audio together. It combines the audio devices into a single audio device but assigns each member to different channels. I have no idea how different applications handle it. It’s up to them. Some choose to mix while others just use the first 2 channels.

Simply you output audio to BlackHole and set BlackHole as the input to your conference app.

But that doesn’t solve the issue that they won’t be able to hear your mic. To fix that you need to mix your mic into BlackHole. Something like GarageBand can do that with input monitoring. Mic input and BlackHole output. You could probably also add your laugh track and other effect in there too.

Devin

**Reply by smktpd** [reply]
Thank you for the explanation.
I actually thought that BlackHole is capable of mixing channels.
I did 2 more tests: selected BlackHole mic and aggregate device that consists of built-in mic + BlackHole mic and in both cases the recipients of the conference can hear the file I play via `mpv laugh.wav --audio-device='coreaudio/BlackHole2ch_UID'` / `mpv laugh.wav --audio-device='coreaudio/~:AMS2_Aggregate:0'`.
Sadly, just like you said, they can't hear me talking into the built-in mic, so my web browser doesn't mix channels automagically.

**Comment by devinroth** [comment]
We developed a paid app called Dipper that also does mixing. Maybe that can help you. https://dipper.audio

Devin
